clang/test/CodeGenSYCL/intel-fpga-no-global-work-offset.cpp

```diff
@@ -1,28 +1,41 @@
-// RUN: %clang_cc1 -fsycl -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
 
 class Foo {
 public:
   [[intel::no_global_work_offset(1)]] void operator()() const {}
 };
 
-template <typename name, typename Func>
-__attribute__((sycl_kernel)) void kernel(const Func &kernelFunc) {
-  kernelFunc();
-}
+template <int SIZE>
+class Functor {
+public:
+  [[intel::no_global_work_offset(SIZE)]] void operator()() const {}
+};
+
+int main() {
+  q.submit([&](handler &h) {
+    Foo boo;
+    h.single_task<class kernel_name1>(boo);
 
-void bar() {
-  Foo boo;
-  kernel<class kernel_name1>(boo);
+    h.single_task<class kernel_name2>(
+        []() [[intel::no_global_work_offset(1)]]{});
 
-  kernel<class kernel_name2>(
-      []() [[intel::no_global_work_offset]]{});
+    h.single_task<class kernel_name3>(
+        []() [[intel::no_global_work_offset(0)]]{});
 
```

Comment:
template support does not accept  [[intel::no_global_work_offset]].

Test is updated since 
valid values are 0 and 1 and attribute parameter is optional,
so [[intel::no_global_work_offset]] means the same as
[[intel::no_global_work_offset(1)]].